api/src/main/java/org/pmiops/workbench/elasticsearch/ElasticSearchService.java

```diff
@@ -0,0 +1,116 @@
+package org.pmiops.workbench.elasticsearch;
+
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.TableResult;
+import jnr.ffi.annotations.Synchronized;
+import org.apache.http.HttpHost;
+import org.elasticsearch.action.admin.cluster.health.ClusterHealthRequest;
+import org.elasticsearch.action.admin.cluster.health.ClusterHealthResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.cohortbuilder.ParticipantCounter;
+import org.pmiops.workbench.cohortbuilder.ParticipantCriteria;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Provider;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+@Service
+public class ElasticSearchService {
+
+  private static final Logger log = Logger.getLogger(ElasticSearchService.class.getName());
+  private RestHighLevelClient client;
+  private Provider<WorkbenchConfig> configProvider;
+  private BigQueryService bigQueryService;
+  private ParticipantCounter participantCounter;
+
+  @Autowired
+  public ElasticSearchService(Provider<WorkbenchConfig> configProvider,
+                              BigQueryService bigQueryService,
+                              ParticipantCounter participantCounter) {
+    this.configProvider = configProvider;
+    this.bigQueryService = bigQueryService;
+    this.participantCounter = participantCounter;
+  }
+
+  /**
+   * This method will generate counts per the
+   * org.pmiops.workbench.model.SearchRequest
+   * using elasticsearch with BigQuery being
+   * the fallback.
+   *
+   * @param userRequest
+   */
+  public Long elasticCount(org.pmiops.workbench.model.SearchRequest userRequest) {
+    //For now if enableElasticsearchBackend is true
+    //we get the cluster name, status and log it
+    //then fallback to bigquery count.
+    //TODO: Implement this method to call either getElasticCount()
+    //or getBigQueryCount() based on certain condition(s) like -> if it's
+    //temporal or not
+    if (configProvider.get().elasticsearch.enableElasticsearchBackend) {
```

Comment:
I think I'd just move this up into the API controller and let this service handle Elastic-only.